### PR TITLE
Add support for Simple Repository API (PEP 503 / PEP 691)

### PR DESCRIPTION
**Pull request by ryanking13**
This adds support for Simple Repository API including [PEP 691](https://peps.python.org/pep-0691/) and [PEP 503](https://peps.python.org/pep-0503/).
This PR also changes the default index URL from `https://pypi.org/pypi/{package_name}/json` to `https://pypi.org/simple`.

I added some tests that send requests to remote package indexes like PyPI, so we can ensure that micropip "works" with famous package indexes. Those tests don't run by default in CI, but one can run them manually by adding `--remote-remote-index-tests` flag or by clicking the button in GHA.

- [ ] changelog

### CODE REVIEW

File: micropip/package_index.py
Comment (rth) [inline micropip/package_index.py:17]: Can you confirm we are never actually downloading the top level index here, because it's >3MB compressed in the case of PyPI.

**Review by rth — APPROVED** [review]
Very nice, thanks @ryanking13 ! I have only one minor comment otherwise LGTM. 

I think once this is merged we should do a micropip release with it. Unless there were other PRs you wanted to do in this area.

File: micropip/package_index.py
Comment (ryanking13) [reply micropip/package_index.py:17]: Yes, I can confirm that we are not downloading the top level index, unless someone changes the index URL to `https://pypi.org` and do `micropip.install("simple")` :)

**Comment by ryanking13** [comment]
> I think once this is merged we should do a micropip release with it. Unless there were other PRs you wanted to do in this area.

Yes, I'll make a release so we can include it in Pyodide 0.24.

**Pull request merged**
